### ISSUE TITLE
Unable to select VMRule from all namespaces with selectAllByDefault

### ISSUE THREAD

**Issue by flokli** [VictoriaMetrics/operator#394]
I deployed victoria-metrics-operator v0.21.0 via the helm chart.

I created a `PrometheusRule` resource in the `prometheus-rules` namespace.

Due to the conversion, the operator created a `VMRule` in the same namespace.

I have a `VMAlert` resource in another namespace, with `selectAllByDefault: true`.

```
❯ kubectl get vmalert -A -oyaml
apiVersion: v1
items:
- apiVersion: operator.victoriametrics.com/v1beta1
  kind: VMAlert
  metadata:
    creationTimestamp: "2021-12-21T11:14:03Z"
    finalizers:
    - apps.victoriametrics.com/finalizer
    generation: 2
    name: vmalert
    namespace: other-namespace
    resourceVersion: "9167521"
    uid: 7741c31d-f1ae-430b-abe9-fea657ea56d9
  spec:
    datasource:
      basicAuth:
        password:
          key: password
          name: vmalert-basic-auth
        username:
          key: username
          name: vmalert-basic-auth
      url: https://$url/prometheus
    image: {}
    notifier:
      url: http://127.0.0.1:9093
    remoteRead:
      basicAuth:
        password:
          key: password
          name: vmalert-basic-auth
        username:
          key: username
          name: vmalert-basic-auth
      url: https://$url/prometheus
    remoteWrite:
      basicAuth:
        password:
          key: password
          name: vmalert-basic-auth
        username:
          key: username
          name: vmalert-basic-auth
      url: https://$url/prometheus
    replicaCount: 1
    resources: {}
    selectAllByDefault: true
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

However, it doesn't seem to pick up the `VMRule` resource in that other namespace.

I port-forwarded the vmalert pod locally, and the only alerting rule I can see (in `/api/v1/groups`) seems to be the pre-configured one, `rate(vmalert_remotewrite_errors_total[1m]) > 0`.

Back when `VMAlert` was in the same namespace as the `PrometheusRule` resource, and `selectAllByDefault` not set, the rules were picked up.

**Comment by flokli** [comment]
In the victoria-metrics-operator, I see the following log entry:

```
{"level":"info","ts":1640089238.8962657,"logger":"factory","msg":"selected Rules","rules":"","namespace":"other-namespace","vmalert":"vmalert"}
```

Which makes me think even with `selectAllByDefault` being set, it still only selects from the current namespace.

**Comment by f41gh7** [comment]
It's regression at 0.21.0 release, will be fixed soon.

**Comment by flokli** [comment]
I see this has been fixed in https://github.com/VictoriaMetrics/operator/pull/396, but no 0.21.1 release has been made yet. Can make a new release, so I can easily apply the fix?

**Comment by f41gh7** [comment]
In next few days will be new release, i suppose at monday. need to fix few issues for it.

**Comment by f41gh7** [comment]
Fix added to the 0.22.0 release
